tests/data_layer/test_data_array.py

```diff
@@ -273,3 +274,40 @@ def test_scalar(self):
 
         df_from_da = da.as_df()
         pd.testing.assert_frame_equal(df_from_da, df)
+
+
+class TestMissingData:
+
+    def test_missing_data_raises(self, small_da):
+        """Should create a pandas.DataFrame from a DataArray
```

Comment:
Comment - 'Should check for NaNs and raise `SmifDataMismatchError` if any'?

Reply:
Just wondering about the error - should we have a `SmifDataMissingError`? I'm not sure if 'mismatch' quite fits, as it were.